src/ngraph/runtime/gpu/gpu_backend.cpp

```diff
@@ -91,13 +91,18 @@ runtime::gpu::GPU_Backend::BackendContext::BackendContext()
 
 void runtime::gpu::GPU_Backend::BackendContext::prepare_runtime_context()
 {
-    //set context current each time in case thread changed
-    m_cuda_manager->SetContextCurrent();
+    // set context current each time in case thread changed
+    bind_cuda_to_thread();
     // add pointers to gpu primitives into the gpu runtime context
     m_runtime_context->gpu_primitives = m_primitive_emitter->get_primitives().data();
     m_runtime_context->gpu_memory_primitives = m_primitive_emitter->get_memory_primitives().data();
 }
 
+void runtime::gpu::GPU_Backend::BackendContext::bind_cuda_to_thread()
```

Comment:
nit: maybe call `bind_cuda_context_to_thread`.

Reply:
Agreed that is more clear, I'll make the change.